pkg/controller/ovn_ic.go

```diff
@@ -167,15 +167,25 @@ func (c *Controller) removeInterConnection(azName string) error {
 
 	if azName != "" {
 		if err := c.OVNNbClient.DeleteLogicalSwitchPorts(nil, func(lsp *ovnnb.LogicalSwitchPort) bool {
-			names := strings.Split(lsp.Name, "-")
-			return len(names) == 2 && names[1] == azName && strings.HasPrefix(names[0], "ts")
+			firstIndex := strings.Index(lsp.Name, "-")
```

Comment:
Add a comment with an example to show why this is needed

Reply:
done

Reply:
done